docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,238 @@
+particular, it may enable `Proxy ARP`_, `Proxy NDP`_, and IP forwarding for
+interfaces that it manages.
+
+.. _Proxy ARP: http://en.wikipedia.org/wiki/Proxy_ARP
+.. _Proxy NDP: http://en.wikipedia.org/wiki/Neighbor_Discovery_Protocol
+
+Route Programming
+~~~~~~~~~~~~~~~~~
+
+Felix is responsible for programming routes to the endpoints on its host into
+the Linux kernel FIB. This ensures that packets destined for those endpoints
+that arrive on at the host are forwarded accordingly.
+
+ACL Programming
+~~~~~~~~~~~~~~~
+
+Felix is also responsible for programming ACLs into the Linux kernel. These
+ACLs are used to ensure that only valid traffic can be sent between
+endpoints, and ensure that endpoints are not capable of circumventing
+Calico's security measures. For more on this, see :doc:`security-model`.
+
+State Reporting
+~~~~~~~~~~~~~~~
+
+Felix is responsible for providing data about the health of the network. In
+particular, it reports errors and problems with configuring its host. This data
+is written into etcd, to make it visible to other components and operators of
+the network.
+
+
+.. _calico-orchestrator-plugin:
+
+Orchestrator Plugin
+-------------------
+
+Unlike Felix there is no single 'orchestrator plugin': instead, there are
+separate plugins for each major cloud orchestration platform (e.g. OpenStack,
+Kubernetes). The purpose of these plugins is to bind Calico more tightly into
+the orchestrator, allowing users to manage the Calico network just as they'd
+manage network tools that were built in to the orchestrator.
+
+A good example of an orchestrator plugin is the Calico Neutron ML2 mechanism
+driver. This component integrates with Neutron's ML2 plugin, and allows users
+to configure the Calico network by making Neutron API calls. This provides
+seamless integration with Neutron, and lowers the barrier of entry to using
+Calico.
+
+The orchestrator plugin is repsonsible for the following tasks:
+
+API Translation
+~~~~~~~~~~~~~~~
+
+The orchestrator will inevitably have its own set of APIs for managing
+networks. The orchestrator plugin's primary job is to translate those APIs into
+the Calico etcd data model (see :doc:`etcd-data-model`) to allow Calico to
+perform the appropriate network programming.
+
+Some of this translation will be very simple, other bits may be more complex in
+order to render a single complex operation (e.g. live migration) into the
+series of simpler operations the rest of the Calico network expects.
+
+Feedback
+~~~~~~~~
+
+If necessary, the orchestrator plugin will provide feedback from the Calico
+network into the orchestrator. Examples include: providing information about
+Felix liveness; marking certain endpoints as failed if network setup failed.
+
+
+.. _calico-etcd-component:
+
+etcd
+----
+
+Calico uses etcd as its backing data store. etcd is a distributed key-value
+store that has a focus on consistency, ensuring that Calico can always build an
```

Comment:
Mmm, it's a tricky one. 'Consistent' in this context is a term-of-art, and the sentence "gives me back the same value as I just wrote into it" is too vague to correctly correspond to what 'consistent' means.

Basically, I was trying to straddle the line between saying nothing about what etcd is and dropping into a sermon on the CAP theorem and consensus protocols.

I'm not sure what I should be shooting for here, so some guidance would be appreciated.